libmamba/src/core/package_handling.cpp

```diff
@@ -57,7 +58,104 @@ namespace mamba
         const fs::u8path& m_file;
     };
 
-    static int copy_data(archive* ar, archive* aw)
+    class libarchive_read_raii
+    {
+    public:
+        libarchive_read_raii()
+            : libarchive_read_raii(archive_read_new()){};
+        static libarchive_read_raii read_disk()
+        {
+            return libarchive_read_raii(archive_read_disk_new());
+        }
+
+        ~libarchive_read_raii()
```

Comment:
While having a user-defined destructor deactivates the copy and move operators, could you add them explicitely `= delete;` so that it's clear that it's the intention? (until they are properly implemented later)
This is just for clarity.
Same remark for the other types.

Reply:
hmm not sure I like the amount of boilerplate code that adds...